DOL.WHD.Section14c.Web/src/modules/components/resultsTable/resultsTableController.js

```diff
@@ -81,22 +74,25 @@ module.exports = function(ngModule) {
           $scope.tableWidget.destroy()
           $scope.tableWidget=null
         }
-       setTimeout(() => this.initDatatable(),0)
+       setTimeout(() => this.initDatatable(id),0)
       }
     }
 
-    this.initDatatable();
+    //this.initDatatable();
+    //this.refreshTable(null, $scope.columns);
 
-    $('#example tbody').on('click', 'td.edit-table-entry', function ($event) {
+    $('#' + $attrs.id).children("table").on('click', 'td.edit-table-entry', function ($event) {
+        $event.preventDefault();
         var tr = $(this).closest('tr');
         var row = $scope.tableWidget.row( tr );
-        $scope.$parent.vm.editEmployee(row[0][0], $event);
+        $scope.$parent.vm["edit" + $attrs.id](row[0][0], $event);
```

Comment:
Instead of having an implicit and undocumented expectation that the parent controller has `edit___` and `delete___` methods, can these be bound to the scope with attributes?  E.g., update the directive to include a couple more scope variables:

```js
scope: {
  "results": "=results",
  ...
  "edit": "&edit",
  "delete": "&delete";
  ...
};
```

Then when creating it:

```html
<results-table id="worker" ... edit="scope.editWorker" delete="scope.deleteWorker"></results-table>
```

Then here in the resultsTable controller, you'd just need to call `$scope.edit()`.  That would be more explicit about what this component expects.  You might also need to test that the `edit` and `delete` scope variables are defined before calling them.